File: JAVA_COURSE/KOTLIN/18_6_MultiArray6/src/edu/javacourse/array/MultiArray.java
package edu.javacourse.array;

public class MultiArray
{
    public static void main(String[] args) {
        // Объявим коснтанту для размера массива
        int SIZE = 10;
        // Создаем двумерный массив 
        char[][] graph = new char[SIZE+1][SIZE+1];

        // Цикл по первой размерности (первые квадратные скобки)
        for (int i = 0; i < SIZE/2+1; i++) {
            // Цикл по второй размерности (вторые квадратные скобки)
            for (int j = 0; j < SIZE+1; j++) {
                if ( i == SIZE/2   || j == SIZE/2 - i || j==SIZE/2+i){ 
                    // Это от ДО
                    graph[i][j] = '#';
                } else {
                    graph[i][j] = ' ';
                }
            }
        }

        // Теперь выводим массив на экран
        // Цикл по первой размерности выводит строки
        for (int i = 0; i < SIZE; i++) {
            // Цикл по второй размерности выводит колонки - вывод одной строки
            for (int j = 0; j <= SIZE; j++) {
                // Используем оператор print - без перехода на следующую строку
                System.out.print(graph[i][j]);
            }
            // Переход на следующую строку
            System.out.println();
        }

    }
}

File: JAVA_COURSE/KOTLIN/18_6_MultiArray6/src/kotlincourse/array/MultiArray.kt
package kotlincourse.array

object MultiArray {
    @JvmStatic
    fun main(args: Array<String>) {
        // Объявим коснтанту для размера массива
        val SIZE = 10
        // Создаем двумерный массив
        val graph = Array(SIZE + 1) { CharArray(SIZE + 1) }

        // Цикл по первой размерности (первые квадратные скобки)
        for (i in 0 until SIZE / 2 + 1) {
            // Цикл по второй размерности (вторые квадратные скобки)
            for (j in 0 until SIZE + 1) {
                if (i == SIZE / 2 || j == SIZE / 2 - i || j == SIZE / 2 + i) {
                    // Это от ДО
                    graph[i][j] = '#'
                } else {
                    graph[i][j] = ' '
                }
            }
        }

        // Теперь выводим массив на экран
        // Цикл по первой размерности выводит строки
        for (i in 0 until SIZE) {
            // Цикл по второй размерности выводит колонки - вывод одной строки
            for (j in 0..SIZE) {
                // Используем оператор print - без перехода на следующую строку
                print(graph[i][j])
            }
            // Переход на следующую строку
            println()
        }

    }
}
